File: msgraph-mail/go/utilities/models/single_value_legacy_extended_property.go
package models

import (
    i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55 "github.com/microsoft/kiota/abstractions/go/serialization"
)

type SingleValueLegacyExtendedProperty struct {
    value *string;
}
func (m *SingleValueLegacyExtendedProperty) getValue()(*string) {
    return m.value
}
func (m *SingleValueLegacyExtendedProperty) getFieldDeserializers()(map[string]func(interface{}, i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55.ParseNode)(error), error) {
    return nil
}
func (m *SingleValueLegacyExtendedProperty) serialize(writer i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55.SerializationWriter)(error) {
    return nil
}
func (m *SingleValueLegacyExtendedProperty) setValue(value *string)() {
    m.value = value
}

File: msgraph-mail/php/Users/SingleValueLegacyExtendedProperty.php
<?php

namespace Microsoft\Graph\Users;

use Microsoft\Kiota\Abstractions\Serialization\ParseNode;

class SingleValueLegacyExtendedProperty extends Entity 
{
    /** @var string $value A property value. */
    private string $value;
    
    /**
     * Gets the value property value. A property value.
     * @return string|null
    */
    public function getValue(): ?string {
        return $this->value;
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable>
    */
    public function getFieldDeserializers(): array {
        echo('This is the body of the deserializer.');
    }

    /**
     * Serializes information the current object
     * @param ?ISerializationWriter $writer $writer Serialization writer to use to serialize this model
    */
    public function serialize(?ISerializationWriter $writer): void {
        parent::serialize($writer);
        $writer->writeStringValue('value', $this->value);
    }

    /**
     * Sets the value property value. A property value.
     *  @param string|null $value Value to set for the value property.
    */
    public function setValue(string $value): void {
        $this->value = $value;
    }

}
